jetty-core/jetty-util/src/main/java/org/eclipse/jetty/util/thread/VirtualThreadPool.java

```diff
@@ -0,0 +1,162 @@
+//
+// ========================================================================
+// Copyright (c) 1995 Mort Bay Consulting Pty Ltd and others.
+//
+// This program and the accompanying materials are made available under the
+// terms of the Eclipse Public License v. 2.0 which is available at
+// https://www.eclipse.org/legal/epl-2.0, or the Apache License, Version 2.0
+// which is available at https://www.apache.org/licenses/LICENSE-2.0.
+//
+// SPDX-License-Identifier: EPL-2.0 OR Apache-2.0
+// ========================================================================
+//
+
+package org.eclipse.jetty.util.thread;
+
+import java.util.concurrent.Executor;
+import java.util.concurrent.RejectedExecutionException;
+import java.util.concurrent.ThreadFactory;
+
+import org.eclipse.jetty.util.StringUtil;
+import org.eclipse.jetty.util.VirtualThreads;
+import org.eclipse.jetty.util.annotation.ManagedAttribute;
+import org.eclipse.jetty.util.component.ContainerLifeCycle;
+import org.eclipse.jetty.util.component.Dumpable;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * An implementation of {@link ThreadPool} interface that does not pool, but instead uses {@link VirtualThreads}.
+ */
+public class VirtualThreadPool extends ContainerLifeCycle implements ThreadFactory, ThreadPool, Dumpable, TryExecutor, VirtualThreads.Configurable
+{
+    private static final Logger LOG = LoggerFactory.getLogger(VirtualThreadPool.class);
+
+    private String _name = "vtp-%x".formatted(hashCode());
+    private VirtualThreads.ThreadFactoryExecutor _virtualExecutor;
+    private final AutoLock.WithCondition _joinLock = new AutoLock.WithCondition();
+
+    public VirtualThreadPool()
+    {
+        if (VirtualThreads.getDefaultVirtualThreadsExecutor() == null)
+            throw new IllegalStateException("Virtual Threads not supported");
+    }
+
+    /**
+     * @return the name of this thread pool
+     */
+    @ManagedAttribute("name of the thread pool")
+    public String getName()
+    {
+        return _name;
+    }
+
+    /**
+     * <p>Sets the name of this thread pool, used as a prefix for the thread names.</p>
+     *
+     * @param name the name of this thread pool
+     */
+    public void setName(String name)
+    {
+        if (isRunning())
+            throw new IllegalStateException(getState());
+        if (StringUtil.isBlank(name))
+            throw new IllegalArgumentException("Blank name");
+        _name = name;
+    }
+
+    @Override
+    protected void doStart() throws Exception
+    {
+        _virtualExecutor = VirtualThreads.getNamedVirtualThreadsExecutor(_name);
```

Comment:
This is the "expensive" virtual thread `Executor` (has to concat strings to give the virtual thread a name), so there should be an option to use the default, rather than the named.
Perhaps just `if (_name == null || _name.isBlank())`.